apps/io-func-wallet-solution-backend/src/infra/crypto/signer.ts

```diff
@@ -0,0 +1,77 @@
+import * as t from "io-ts";
+
+import { pipe } from "fp-ts/function";
+import * as A from "fp-ts/lib/Array";
+import * as E from "fp-ts/lib/Either";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as jose from "jose";
+
+import { Signer } from "../../signer";
+import { ECKey, Jwk, RSAKey } from "../../jwk";
+import { validate } from "../../validation";
+import { CryptoConfiguration } from "./config";
+
+const supportedSignAlgorithms = [
+  "ES256",
+  "ES256K",
+  "ES384",
+  "ES512",
+  "RS256",
+  "RS384",
+  "RS512",
+  "PS256",
+  "PS384",
+  "PS512",
+];
+
+export class CryptoSigner implements Signer {
+  #configuration: CryptoConfiguration;
+
+  constructor(cnf: CryptoConfiguration) {
+    this.#configuration = cnf;
+  }
+
+  getPublicKeys = () =>
+    pipe(
+      this.#configuration.jwks,
+      validate(
+        t.array(t.union([t.exact(ECKey), t.exact(RSAKey)])),
+        "JWKs appears to not be a public key array"
+      )
+    );
+
+  // TODO: [SIW-260] Make algorithm management separate and not hard-coded
+  createJwtAndsign = (typ: string) => (payload: jose.JWTPayload) =>
+    pipe(
+      getFirstPrivateKeyByAlg(this.#configuration.jwks, "EC"),
+      TE.fromOption(() => new Error("No keys found for this algorithm")),
+      TE.chain((privateKey) =>
+        pipe(
+          TE.tryCatch(() => jose.importJWK(privateKey), E.toError),
+          TE.map((joseKey) => ({ joseKey, kid: privateKey.kid }))
+        )
+      ),
+      TE.chain(({ joseKey, kid }) =>
+        TE.tryCatch(
+          () =>
+            new jose.SignJWT(payload)
+              .setProtectedHeader({ kid, alg: "ES256", typ })
+              .setIssuedAt()
+              .setExpirationTime("1h")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Should we define `ES256` and `1h` as application configuration? Maybe just the expiration.

Reply:
at the beginning i put a todo to the function to make these parameters dynamic. Maybe we should define a global variable for the expiration date?

Reply:
My biggest concern is: they are business-sensitive information and are sunk deep in code implementation. 

We can put them somewhere at the top, so they are evident; I thought configuration to be the right place (even if we use constant values), but any other place will be fine I think.

Reply:
Ok, I moved these parameters to the configuration!